CommonApplication.cmake

```diff
@@ -122,6 +123,34 @@ function(common_application Name)
     endif()
   endif()
 
+  if(NOT THIS_GUI OR THIS_DOXYGEN)
+    # run binary with --help to capture output for doxygen
+    set(_doc "${PROJECT_BINARY_DIR}/help/${Name}.md")
+    set(_cmake "${CMAKE_CURRENT_BINARY_DIR}/${Name}.cmake")
+    file(WRITE ${_cmake} "
+      execute_process(COMMAND \${APP} --help
+      OUTPUT_FILE ${_doc} ERROR_FILE ${_doc}.error)
+      file(READ ${_doc} _doc_content)
+      if(NOT _doc_content)
+        message(FATAL_ERROR \"${Name} is missing --help\")
+      endif()
+      file(WRITE ${_doc} \"${Name} {#${Name}}
+============
+
+```
+\${_doc_content}
+```
+\")
+")
+    add_custom_command(OUTPUT ${_doc}
+      COMMAND ${CMAKE_COMMAND} -DAPP=$<TARGET_FILE:${Name}> -P ${_cmake}
+      DEPENDS ${Name}
+      COMMENT "Creating help for ${Name}")
+    add_custom_target(${Name}-help ALL DEPENDS ${_doc})
+    add_dependencies(${PROJECT_NAME}-all ${Name}-help)
```

Comment:
Fail early is good in principal, but I agree that building 'help' everytime you build 'all' is wrong, also annoying as I personally do 'all' quite often. Moreover, doxygen dependes on install, not on 'all'!

To fight problems being not noticed we could run doxygen as part of CI and report the errors from error. Now we actually have sometimes doxygen errors/warnings going unnoticed. We catch them only from time to time by checking the documentation CI plans...